sql/select.go

```diff
@@ -432,6 +429,10 @@ func (v *indexInfo) makeConstraints(exprs []parser.Exprs) {
 					if !endDone {
 						constraint.end = c
 					}
+				case parser.NE:
```

Comment:
`!=` expressions are used by `makeSpans` equivalently to `a IS NOT NULL`. I don't want to transform `!=` to `a IS NOT NULL` because that affects other simplifications. I'll add a comment here explaining this.